### DISCUSSION TITLE
magick: missing output filename `-fill'

### DISCUSSION THREAD

**Discussion by Pr0methean** [ImageMagick/ImageMagick#5251] [Help]
I have a shell script that is running
```
magick "/__w/OcHd/OcHd/png/32x32/checksSmall.png" -fill "#d1ba8a" -colorize 100% -background "#dfd5aa" -alpha remove -alpha off "/__w/OcHd/OcHd/tmp/32x32/sand1.png"
```
and getting
```
magick: missing output filename `-fill' at CLI arg 2 @ error/magick-cli.c/ProcessCommandOptions/524.
```
How do I fix this error? Full logs are at https://github.com/Pr0methean/OcHd/runs/6977381865?check_suite_focus=true; script sources are at https://github.com/Pr0methean/OcHd/commit/c75ff62bca5f2de6f5eb102ac56ce4fed6b972f8. The apparent triggering line in `build_size.sh` is line 455, `push checksSmall ${sand_s} sand1 ${sand}`. The ImageMagick version I'm using is:
```
Version: ImageMagick 7.1.0-37 Q16-HDRI x86_64 b65d0d9af:20220605 https://imagemagick.org/
Copyright: (C) 1999 ImageMagick Studio LLC
License: https://imagemagick.org/script/license.php
Features: Cipher DPC HDRI OpenMP(4.5) 
Delegates (built-in): fontconfig freetype heic jbig jng jpeg lcms lzma png tiff webp x xml zlib
Compiler: gcc (9.4)
```
The scripts get this version as part of Docker image ghcr.io/pr0methean/inkscapeandimagemagick:main@sha256:452835a3c47ac6d358b45f642836c063492e76a15d5dc6b6a2c1b926a15b6158. I generated that image today. The Dockerfile is at https://github.com/Pr0methean/InkscapeAndImageMagick/commit/130e5e17aa4f759f1bbf5f8b5696ef8797f32854 and the Docker build logs are at https://github.com/Pr0methean/InkscapeAndImageMagick/runs/6974241176?check_suite_focus=true.

I've also tried including the # sign in the color constants passed to the subroutine scripts, rather than prepending it at substitution time; but that triggers an apparent bug in GNU Parallel that cuts off both the color parameter and all subsequent parameters.

**Comment by fmw42** [comment]
Your command works fine for me on IM 7.1.0.39 Q16 HDRI on Mac OSX Monterey using the lena.png image.  You did not provide your input image. So perhaps a problem with it or your path. Enclosing the hex color in double quotes should have taken care of that issue. Does it work if you remove the % from 100%? You should not need both -alpha remove and -alpha off. Either should be sufficient.

Does your command work in standalone mode rather than from a script? Can you copy and paste into a terminal window and have it run successfully?

**Comment by snibgo** [comment]
If a script builds the command, perhaps the command isn't exactly as you think it is.

I suggest inserting "-debug command" as the first option, before the input file name. That will send some text to stderr. Please paste that text here.
